components/post_view/combined_system_message/combined_system_message.jsx

```diff
@@ -191,117 +192,111 @@ export default class CombinedSystemMessage extends React.PureComponent {
-                    values={{
-                        firstUser,
-                        actor,
-                    }}
+        const options = {
+            atMentions: true,
+            mentionKeys: [{key: firstUser}, {key: secondUser}, {key: actor}],
+            mentionHighlight: false,
+        };
+
+        if (numOthers > 1) {
+            return (
+                <LastUsers
+                    actor={actor}
+                    expandedLocale={postTypeMessage[postType].many_expanded}
+                    formatOptions={options}
+                    postType={postType}
+                    usernames={usernames}
                 />
             );
+        }
+
+        let localeHolder;
+        if (numOthers === 0) {
+            localeHolder = {
+                id: postTypeMessage[postType].one.id,
+                defaultMessage: postTypeMessage[postType].one.defaultMessage,
+            };
 
             if (
                 (userIds[0] === this.props.currentUserId || userIds[0] === this.props.currentUsername) &&
                 postTypeMessage[postType].one_you
             ) {
-                formattedMessage = (
-                    <FormattedHTMLMessage
-                        id={postTypeMessage[postType].one_you.id}
-                        defaultMessage={postTypeMessage[postType].one_you.defaultMessage}
-                        values={{
-                            actor,
-                        }}
-                    />
-                );
+                localeHolder = {
+                    id: postTypeMessage[postType].one_you.id,
+                    defaultMessage: postTypeMessage[postType].one_you.defaultMessage,
+                };
             }
         } else if (numOthers === 1) {
-            formattedMessage = (
-                <FormattedHTMLMessage
-                    id={postTypeMessage[postType].two.id}
-                    defaultMessage={postTypeMessage[postType].two.defaultMessage}
-                    values={{
-                        firstUser,
-                        secondUser: userDisplayNames[1],
-                        actor,
-                    }}
-                />
-            );
-        } else if (numOthers > 1) {
-            formattedMessage = (
-                <LastUsers
-                    actor={actor}
-                    expandedLocale={postTypeMessage[postType].many_expanded}
-                    postType={postType}
-                    userDisplayNames={userDisplayNames}
-                />
-            );
+            localeHolder = {
+                id: postTypeMessage[postType].two.id,
+                defaultMessage: postTypeMessage[postType].two.defaultMessage,
+            };
         }
 
-        return formattedMessage;
+        const formattedMessage = formatMessage(localeHolder, {firstUser, secondUser, actor});
+        const formattedText = formatText(formattedMessage, options).replace(/p>/g, 'span>');
```

Comment:
This string replacement is an ugly hack to fix vertical spacing.  Basically changing `p` to `span` element.

Reply:
@saturninoabril If you use the Markdown component with the singleline option, does that fix the vertical spacing?

Reply:
I tried it first but had the same spacing issue.

Reply:
Oh wait, I missed the singleline option.  I'll find time to try it.

Reply:
Thanks @hmhealey!  Using Markdown with singleline option fixes the spacing.